nexus/src/app/background/common.rs

```diff
@@ -0,0 +1,257 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Common facilities for background tasks
```

Comment:
I'm especially interested in this module - I'd like to leverage this for service rebalancing, updates, periodic system monitoring, etc.

Lemme know when it's ready for me to take a look -- if we're searching for subcomponents of this PR to split out, I'd happily review this first.

Reply:
Will do.

The approach I've taken here is quite lightweight: there's a trait with one function called `activate()` and you impl that.  Then the subsystem creates and registers all of the impls of this trait (see `init()`).  Then the thing can be activated on some time interval or explicitly by other parts of Nexus.  There's some simple support for causing one of these to activate another one.  The `DnsPropagator` one uses this.  This is the `GenericWatcher` stuff.

There's a lot this _doesn't_ do, but I really wanted to avoid overengineering it ahead of time.